### PR TITLE
fix: pbft gas validation in PbftSyncPacketHandler

### PR DESCRIPTION
**Pull request by kstdl**

### CODE REVIEW

**Review by MatusKysel — APPROVED** [review]

**Review by mfrankovi — APPROVED** [review]

File: libraries/core_libs/consensus/src/pbft/pbft_manager.cpp [hunk 1]
```diff
@@ -1905,6 +1905,17 @@ std::optional<SyncBlock> PbftManager::processSyncBlock() {
...
   }
 
+  auto prev_pbft_block = pbft_chain_->getPbftBlockInChain(last_pbft_block_hash);
+
+  std::vector<blk_hash_t> ghost;
+  dag_mgr_->getGhostPath(prev_pbft_block.getPivotDagBlockHash(), ghost);
+  if (ghost.size() > 1 && sync_block.first.pbft_blk->getPivotDagBlockHash() != ghost[1]) {
+    if (!checkBlockWeight(sync_block.first)) {
+      LOG(log_er_) << "processSyncBlock: SyncBlock block " << pbft_block_hash << " is overweighted";
+      net->handleMaliciousSyncPeer(sync_block.second);
```
Comment (MatusKysel) [inline libraries/core_libs/consensus/src/pbft/pbft_manager.cpp:1915]: missing `return std::nullopt;`

Comment (MatusKysel) [reply libraries/core_libs/consensus/src/pbft/pbft_manager.cpp:1915]: and also `sync_queue_.clear();`

**Review by MatusKysel — APPROVED** [review]

File: libraries/core_libs/consensus/src/pbft/pbft_manager.cpp [hunk 2]
```diff
@@ -1905,6 +1905,17 @@ std::optional<SyncBlock> PbftManager::processSyncBlock() {
     return std::nullopt;
   }
 
+  auto prev_pbft_block = pbft_chain_->getPbftBlockInChain(last_pbft_block_hash);
```
[the remark this answers is not in the document]
Comment (kstdl) [reply libraries/core_libs/consensus/src/pbft/pbft_manager.cpp:1908]: What do you mean by this?

[the remark this answers is not in the document]
Comment (MatusKysel) [reply libraries/core_libs/consensus/src/pbft/pbft_manager.cpp:1908]: few lines above

**Pull request merged**